arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedParquetDefinitionLevelReader.java

```diff
@@ -46,122 +47,217 @@ public VectorizedParquetDefinitionLevelReader(
     super(bitWidth, maxDefLevel, readLength, setArrowValidityVector);
   }
 
-  abstract class NumericBaseReader {
-    public void nextBatch(
+  abstract class CommonBaseReader {
+    protected void nextCommonBatch(
```

Comment:
why not leave that as `nextBatch()`? Does the `common` in the name help in any way?

Reply:
We have two public methods `nextBatch` and `nextDictEncodedBatch`. This one is the implementation and does not need to be exposed. I named it this way as it is common to both `nextBatch` and `nextDictEncodedBatch`; giving it the same name as one of those two seems more confusing to me.